Objects/listobject.c

```diff
@@ -1253,45 +1253,50 @@ list_extend_iter(PyListObject *self, PyObject *iterable)
     return -1;
 }
 
+static int
+list_extend_lock_held(PyListObject *self, PyObject *iterable)
+{
+    iterable = PySequence_Fast(iterable, "argument must be iterable");
+    if (!iterable) {
+        return -1;
+    }
+
+    int res = list_extend_fast(self, iterable);
+    Py_DECREF(iterable);
```

Comment:
I'd prefer to use a local variable, rather than mutating the argument.

```suggestion
    PyObject *seq = PySequence_Fast(iterable, "argument must be iterable");
    if (!seq) {
        return -1;
    }

    int res = list_extend_fast(self, seq);
    Py_DECREF(seq);
```